data/moves.js

```diff
@@ -1294,7 +1294,7 @@ exports.BattleMovedex = {
 				if (move.id === 'skyuppercut' || move.id === 'thunder' || move.id === 'hurricane' || move.id === 'smackdown' || move.id === 'thousandarrows' || move.id === 'helpinghand') {
 					return;
 				}
-				if (source.hasAbility('noguard') || target.hasAbility('noguard')) {
+				if (source.hasAbility('noguard') || target.hasAbility('noguard') || target.volatiles['lockon']) {
```

Comment:
If it's going to be done this way, `target.volatiles['lockon']` should be `(target.volatiles['lockon'] && target.volatiles['lockon'].source === source)` to make sure the user of the move is the one that used Lock-On or Mind Reader last turn (plus a similar check in Lock-On itself).